### PR TITLE
[ISSUE #77] optimize process of allocate opaque by atomic without lock.

### PR DESCRIPTION
**Pull request by ifplusor**
#77 replace lock with fetch_add(1) when allocate opaque

### CODE REVIEW

File: src/protocol/RemotingCommand.cpp [hunk 1]
```diff
@@ -32,11 +33,9 @@ RemotingCommand::RemotingCommand(int code,
...
       m_remark(""),
       m_pExtHeader(pExtHeader) {
-  boost::lock_guard<boost::mutex> lock(m_clock);
-  m_opaque = (s_seqNumber.load(boost::memory_order_acquire)) %
-             (numeric_limits<int>::max());
-  s_seqNumber.store(m_opaque, boost::memory_order_release);
-  ++s_seqNumber;
+
+  // mask sign bit
+  m_opaque = s_seqNumber.fetch_add(1, boost::memory_order_acq_rel) & numeric_limits<int>::max();
```
Comment (lizhanhui) [inline src/protocol/RemotingCommand.cpp]: As we only need an atomic incremental number and no memory barrier is required, boost::memory_order_relaxed should be fine here.

Comment (ifplusor) [reply src/protocol/RemotingCommand.cpp]: yeah. yesterday, i change this to 'memory_order_acq_rel' when i try to fix other bug, but that have no relation to memory order. i have revert it.

**Comment by lizhanhui** [comment]
+1 to get this pull request merged.

**Comment by ifplusor** [comment]
close #77

**Pull request merged**